state/factory/statedb.go

```diff
@@ -131,8 +85,30 @@ func DisableWorkingSetCacheOption() StateDBOption {
 	}
 }
 
+// CreateTrieDB creates state db from config
+func CreateTrieDB(dbCfg db.Config, chainCfg blockchain.Config) (db.KVStore, error) {
+	dbPath := chainCfg.TrieDBPath
+	if len(dbPath) == 0 {
+		return nil, errors.New("Invalid empty trie db path")
+	}
+	dbCfg.DbPath = dbPath
+
+	return db.NewBoltDB(dbCfg), nil
+
+}
+
+// CreateTrieDBWithCache creates state db with cache from config
+func CreateTrieDBWithCache(dbCfg db.Config, chainCfg blockchain.Config) (db.KVStore, error) {
```

Comment:
`CreateTrieDBWithCache`
and also use `dbPath string` as input

Reply:
not only `chainCfg.TrieDBPath`, but also  `chainCfg.StateDBCacheSize` was used.